src/main/scala/no/ndla/imageapi/controller/ImageControllerV2.scala

```diff
@@ -64,51 +99,43 @@ trait ImageControllerV2 {
 
     val getImages =
       (apiOperation[SearchResult]("getImages")
-        summary "Show all images"
-        notes "Shows all the images in the ndla.no database. You can search it too."
+        summary "Find images"
+        notes "Find images in the ndla.no database."
         parameters(
-        headerParam[Option[String]]("X-Correlation-ID").description("User supplied correlation-id. May be omitted."),
-        queryParam[Option[String]]("query").description("Return only images with titles, alt-texts or tags matching the specified query."),
-        queryParam[Option[String]]("minimum-size").description("Return only images with full size larger than submitted value in bytes."),
-        queryParam[Option[String]]("language").description("The ISO 639-1 language code describing language used in query-params."),
-        queryParam[Option[String]]("license").description("Return only images with provided license."),
-        queryParam[Option[String]]("includeCopyrighted").description("Return copyrighted images. May be omitted."),
-        queryParam[Option[String]]("sort").description(
-          """The sorting used on results.
-             Default is by -relevance (desc) when querying.
-             When browsing, the default is title (asc).
-             The following are supported: relevance, -relevance, title, -title, lastUpdated, -lastUpdated, id, -id""".stripMargin),
-        queryParam[Option[Int]]("page").description("The page number of the search hits to display."),
-        queryParam[Option[Int]]("page-size").description("The number of search hits to display for each page.")
+        asHeaderParam[Option[String]](correlationId),
+        asQueryParam[Option[String]](query),
+        asQueryParam[Option[Int]](minSize),
+        asQueryParam[Option[String]](language),
+        asQueryParam[Option[String]](license),
+        asQueryParam[Option[Boolean]](includeCopyrighted),
+        asQueryParam[Option[String]](sort),
+        asQueryParam[Option[Int]](pageNo),
+        asQueryParam[Option[Int]](pageSize)
       )
         authorizations "oauth2"
         responseMessages response500)
 
     get("/", operation(getImages)) {
-      val minimumSize = intOrNone("minimum-size")
-      val query = paramOrNone("query")
-      val language = paramOrNone("language")
-      val license = params.get("license")
-      val pageSize = intOrNone("page-size")
-      val page = intOrNone("page")
-      val sort = Sort.valueOf(paramOrDefault("sort", ""))
-      val includeCopyrighted = booleanOrDefault("includeCopyrighted", false)
+      val minimumSize = intOrNone(this.minSize.param_name)
```

Comment:
Liker veldig godt at parameternavn nå blir vanskeligere å være inkonsistent mellom dok og selve implementasjon. :+1: Kjenner jeg fikk litt "java-vibber" når jeg så `this` :p Hadde det vært mulig å kalle parameter-definisjonene noe annet, f.eks `minSizeParam.name`? Da slipper vi `this`

En annen idé hadde kanskje vært å utvide `Param` klassen til å hente parameteret fra selve requesten også. Da kunne denne sett slik ut: `val minimumSize = minSizeParam.value.map(toInt)`. Fremdeles bare en tanke. Litt usikker enda på om jeg liker den eller ikke

Reply:
Hehe. 

Jeg også var på vei inn i sistnevnte variant. Da mere `val minimumSizeValue = minSize.value` og heller bestemme i definisjonen at det skal være `Int` og om det skal være optional, default eller annet. Dermed kunne jeg deretter inlinet `minimumSizeValue` og forholdt meg bare til ett navn... Men det ble litt for omfattende ift. hva jeg fikk til. 

Så når jeg slo fra meg det, startet jeg også med å ha `Param` etter alle navnene for å unngå `this`. Men følte det ble veldig klønete variabelnavn, og at "totalopplevelsen" ble bedre med `this`. Den er jo strengt tatt unødvendig i en del tilfeller også, men har gjort det konsekvent for lesbarhet.

Sterkt uenig?

Reply:
Skjønner. Nope, kan beholde det slik det er 👍